frontend/vite-project/src/pages/PropertyOverviewPage.jsx

```diff
@@ -1,31 +1,167 @@
+      setSqft(sqft);
+      setPurchaseWorksheet(purchase_worksheet);
+      setProperty(response.data)
+      if (portfolio) {
+        setIsInPortfolio(true)
+      };
+      if (list_of_properties) {
+        setIsInList(true)
+      };
+    };
+
+    const getRents = async () => {
+      const response = await api.get(`rent/${propertyId}/`)
+
+      const {max, mean, median, min} = response.data;
+      setMax(max);
+      setMean(mean);
+      setMedian(median);
+      setMin(min);
+    }
+
+    const handlePortfolioToggle = async () => {
+      if (isInPortfolio) {
+        try {
+          const response = await api.put(`properties/remove/property/${propertyId}/portfolio/`);
+          setIsInPortfolio(false);
+        } catch(error) {
+          console.log(error)
+        }
+      } else {
+        try {
+          const response = await api.put(`properties/add/property/${propertyId}/portfolio/`);
+          setIsInPortfolio(true)
+        } catch(error) {
+          console.log(error)
+        }
+      }
+    };
+
+  const getUserLists = async () => {
+
+    try {
+      const response = await api.get('properties/lists/')
+      setUserLists(response.data)
+    } catch(error) {
+      console.log(error)
+    }
+
+  }
+  
+  useEffect(() => {
+    getUserLists();
+  }, [user]);
+    
+    const handleListSelection = (listId) => {
+      setSelectedListId(listId);
+    };
+
+  const handleAddToList = async () => {
+    if (selectedListId) {
+      try {
+        const response = await api.put(`properties/add/property/${propertyId}/lists/${selectedListId}/`)
+
+      } catch(error) {
+        console.log(error)
+      }
+    }
+  }
+
+    const getImage = async () => {
+      if (property !== null){
+        try {
+          const response = await api.get(`map/propertyimage/${encodeURIComponent(address)}`, {
+              responseType: 'blob', // Set response type to 'blob'
+          });
+          
+          const imageBlob = response.data;
+          const imageUrl = URL.createObjectURL(imageBlob);
+
+          setNewPropertyImage(imageUrl);
+        } catch (error) {
+            console.error("Error fetching property image:", error);
+        }
+      }
+    }
+
+    useEffect(() => {
+      getProperty();
+    }, [user]);
+
+    useEffect(() => {
+      getImage();
+      getRents();
+    }, [property]);
 
       return (
         <div className="aProperty">
-          <PropertyComp property={property}/>
+          <div>
+        <PropertyComp address={address} newPropertyImage={newPropertyImage}
+        beds={beds} baths={baths} sqft={sqft} max={max} mean={mean} median={median}
+         min={min} details={details} />
+         </div>
+        <div>
+            <button onClick={handlePortfolioToggle}>
+                    {isInPortfolio ? "Remove from Portfolio" : "Add to Portfolio"}
+            </button>
+  
+              <div>
+                <button onClick={handleAddToList}>Add to List</button>
```

Comment:
This file is huge and very hard to keep up with everything going on within the page. It may be a good idea to break it up onto smaller components to make your code more readable